network/src/peer_store/db.rs

```diff
@@ -186,6 +186,27 @@ impl PeerAddr {
     }
 }
 
+pub fn get_random_peers(conn: &Connection, count: u32) -> DBResult<Vec<(PeerId, Multiaddr)>> {
+    // random select peers
+    let mut stmt = conn.prepare("SELECT id, peer_id FROM peer_info WHERE ban_time < strftime('%s','now') ORDER BY RANDOM() LIMIT :count")?;
+    let rows = stmt.query_map_named(&[(":count", &count)], |row| {
+        (
+            row.get::<_, u32>(0),
+            PeerId::from_bytes(row.get(1)).expect("parse peer_id"),
+        )
+    })?;
+
+    let mut peers = Vec::with_capacity(count as usize);
+    for row in rows {
+        let (id, peer_id) = row?;
+        let mut addrs = PeerAddr::get_addrs(conn, id, 1)?;
+        if let Some(addr) = addrs.pop() {
+            peers.push((peer_id, addr));
+        }
+    }
+    Ok(peers)
```

Comment:
Will get different peers to the same address. The address is discovered from different peers.